### PR TITLE
Align Content-Length handling with libcurl

### LINKED ISSUE #36: Align Content-Length behavior with libcurl
Content-Length handling for incoming messages in `BodyReader::header_defined` (`src/body.rs`) diverges from RFC 9110 §8.6 / RFC 9112 §6.3 and from libcurl. This issue documents the exact rules we want; the implementation is a follow-up PR.

## Current behavior

- `header_defined` reads Content-Length through a `Fn(HeaderName) -> Option<&str>` lookup backed by `HeaderMap::get`, which returns only the **first** header line. The `TooManyContentLengthHeaders` check inside the function can never trigger, so `Content-Length: 5` followed by `Content-Length: 100` silently frames the body on 5 and desyncs a keep-alive connection.
- Values are parsed with `str::parse::<u64>`, which accepts a leading `+` (`+42` parses as 42). RFC 9110 requires `1*DIGIT`.
- A comma-separated list (`42, 42`) fails to parse and is rejected. That is permitted by the RFC, but stricter than libcurl and browsers.
- The outgoing paths (`src/client/amended.rs`, `src/server/amended.rs`) count `get_all` and reject any repeat, but use the same `+`-accepting parse.

This affects both incoming paths, since they share `header_defined`: client responses via `BodyReader::for_response` and server requests via `BodyReader::for_request`.

## Rules we want

These match libcurl's `lib/http.c` (the block beginning "Check for Content-Length: header lines to get size. Browsers insist we should accept multiple Content-Length headers and that a comma separated list also is fine and then we should accept them all as long as they are the same value. Different values trigger error.") and RFC 9110 §5.3, §8.6, RFC 9112 §6.3.

1. **Grammar.** A single value is `1*DIGIT`. Leading zeros are allowed. No sign, no internal whitespace, no other characters. Leading and trailing whitespace is already stripped by the header parser. Anything else is `Error::BadContentLengthHeader`.
2. **Repeats are lists.** Multiple `Content-Length` header lines are equivalent to one line with the values joined by commas (RFC 9110 §5.3). Both forms are handled identically: collect all values, split each on `,`, trim each element, apply rule 1 to each element.
3. **All values must agree.** If every element parses and all are numerically equal, the message uses that single value. `042` and `42` are equal. If any two elements differ, the message is rejected with `Error::TooManyContentLengthHeaders`. There is no first-wins or last-wins.
4. **Empty is invalid.** `Content-Length:` with no value, or an empty list element (`42,,42`, a trailing comma), is `Error::BadContentLengthHeader`.
5. **Overflow is invalid.** A value that does not fit in `u64` is `Error::BadContentLengthHeader`. libcurl instead logs it, ignores the value and marks the connection to close, but that exists to cope with a 32-bit `curl_off_t`. With `u64` any such value is garbage. This is the one deliberate divergence.
6. **One parser everywhere.** The outgoing paths in `client/amended.rs` and `server/amended.rs` keep rejecting repeated header lines outright (we control what we send), but must use the same value parser so a user-supplied `+42` is rejected there too.

## Implementation notes

- The `header_lookup` closure passed to `header_defined` needs to expose all values of a header, e.g. by taking the `HeaderMap` or an iterator over `get_all`. `BodyReader` is `pub(crate)`, so the signature change is free.
- One shared `parse_content_length` helper taking an iterator of values and returning `Result<Option<u64>, Error>`, used by `header_defined` and both `amended.rs` analyzers.

## Test cases

| Input | Result |
|---|---|
| `Content-Length: 42` | 42 |
| `Content-Length: 042` | 42 |
| `Content-Length: 42` + `Content-Length: 42` | 42 |
| `Content-Length: 42, 42` | 42 |
| `Content-Length: 42` + `Content-Length: 042` | 42 |
| `Content-Length: 42` + `Content-Length: 43` | `TooManyContentLengthHeaders` |
| `Content-Length: 42, 43` | `TooManyContentLengthHeaders` |
| `Content-Length: +42` | `BadContentLengthHeader` |
| `Content-Length: -1` | `BadContentLengthHeader` |
| `Content-Length: 4 2` | `BadContentLengthHeader` |
| `Content-Length: 42abc` | `BadContentLengthHeader` |
| `Content-Length:` (empty) | `BadContentLengthHeader` |
| `Content-Length: 42,` | `BadContentLengthHeader` |
| `Content-Length: 99999999999999999999999` | `BadContentLengthHeader` |

Both the client response path and the server request path should be covered by tests.

## References

- RFC 9110 §5.3 (repeated field lines), §8.6 (Content-Length): https://www.rfc-editor.org/rfc/rfc9110#section-8.6
- RFC 9112 §6.3 (message body length): https://www.rfc-editor.org/rfc/rfc9112#section-6.3
- libcurl response Content-Length handling: https://github.com/curl/curl/blob/master/lib/http.c
- libcurl number parsing, `str_num_base` requires the first character to be a digit: https://github.com/curl/curl/blob/master/lib/curlx/strparse.c

### PR DESCRIPTION
**Pull request by algesten**
## Summary

Content-Length handling now follows RFC 9110 §8.6 and RFC 9112 §6.3 the same way libcurl does.

- A value is `1*DIGIT`. Leading zeros are allowed. A sign, whitespace inside the number or any other character is `BadContentLengthHeader`. Rust's integer parsing used to accept `+42`.
- Repeated header lines and comma separated lists are treated identically: every element must parse and all must be the same number, `42` and `042` included. Differing values are `TooManyContentLengthHeaders`. Previously the incoming response and request paths read only the first line through `HeaderMap::get`, so the duplicate check there was unreachable and `Content-Length: 5` followed by `Content-Length: 100` framed the body on 5.
- An empty value, an empty list element and a number that does not fit in `u64` are `BadContentLengthHeader`.

A single `parse_content_length` in `body.rs` implements the rules. `BodyReader::for_request` and `for_response` take the `HeaderMap` instead of a first-value lookup closure, and the outgoing analyzers in `client/amended.rs` and `server/amended.rs` apply the single-value grammar (`parse_content_length_value`): they keep rejecting repeated header lines outright, and a comma separated list is rejected too, since a sender must only emit `1*DIGIT`. Transfer-Encoding handling is unchanged.

Tests: unit tests for the parser covering the full table from the issue, client response tests (identical duplicates accepted, differing lines and a differing list rejected, sign rejected), server request tests for the same cases, outgoing tests for a signed and for a comma separated Content-Length on a client request and on a server response, and wire-level tests for empty values.

Closes #36

### CODE REVIEW

File: src/client/amended.rs [hunk 1]
```diff
@@ -169,15 +169,8 @@ impl AmendedRequest {
...
-        if let Some(h) = self.headers_get(header::CONTENT_LENGTH) {
-            let n = h
-                .to_str()
-                .ok()
-                .and_then(|s| s.parse::<u64>().ok())
-                .ok_or(Error::BadContentLengthHeader)?;
-            content_length = Some(n);
-        }
+        let content_length =
+            parse_content_length(self.headers_get(header::CONTENT_LENGTH).into_iter())?;
```
Comment (algesten) [inline src/client/amended.rs]: **P1: an outgoing comma separated `Content-Length` is now accepted and written to the wire verbatim.**

`parse_content_length` splits every value on `,`, so this call accepts a user supplied `Content-Length: 42, 42` (any list of equal numbers). On `main` the same value failed `str::parse::<u64>` and was rejected with `BadContentLengthHeader`. On this branch `analyze` returns `content_length = Some(42)`, and `do_write_headers` in `sendreq.rs` copies the `HeaderValue` bytes unchanged, so the request is sent as:

```
POST / HTTP/1.1\r\nhost: example.com\r\ncontent-length: 42, 42\r\n\r\n
```

RFC 9110 §8.6 defines `Content-Length = 1*DIGIT` and says a sender MUST NOT send a value that does not match that ABNF; the `42, 42` tolerance in the same section is for recipients only. Issue #36 rule 6 keeps this path rejecting repeated lines because "we control what we send", and that reasoning applies equally to the list form. This is a silent loosening compared to `main` that the PR description does not mention. The incoming paths should stay list tolerant; the two outgoing analyzers should apply the single value grammar only (for example call `parse_content_length_value` on the one value, or reject a `,` before handing the value to the list parser).

Regression test: `error::tests_client::test_list_content_length_header` in 6dd1b37. It fails on this branch and passes on `main`. It accepts either `BadContentLengthHeader` or `TooManyContentLengthHeaders`, since the issue does not pin the variant for a sender side list.

File: src/server/amended.rs [hunk 2]
```diff
@@ -68,15 +68,8 @@ impl AmendedResponse {
...
-        if let Some(h) = self.headers_get(header::CONTENT_LENGTH) {
-            let n = h
-                .to_str()
-                .ok()
-                .and_then(|s| s.parse::<u64>().ok())
-                .ok_or(Error::BadContentLengthHeader)?;
-            content_length = Some(n);
-        }
+        let content_length =
+            parse_content_length(self.headers_get(header::CONTENT_LENGTH).into_iter())?;
```
Comment (algesten) [inline src/server/amended.rs]: **P1: same as the client analyzer: an outgoing comma separated `Content-Length` is now accepted and written to the wire verbatim.**

A user supplied response header `content-length: 42, 42` passes this call with `content_length = Some(42)` on this branch, where `main` rejected it with `BadContentLengthHeader` because `"42, 42".parse::<u64>()` fails. The header bytes are then copied unchanged by the response writer, so the server emits:

```
HTTP/1.1 200 OK\r\ncontent-length: 42, 42\r\n\r\n
```

A sender MUST NOT send a `Content-Length` that does not match `1*DIGIT` (RFC 9110 §8.6). The `count_len > 1` check above keeps rejecting repeated lines for exactly that reason, so the list form should be rejected here too; only the incoming `BodyReader` paths should be list tolerant.

Regression test: `server::test::state_provide_response::provide_response_with_content_length_list_is_rejected` in 6dd1b37. It fails on this branch and passes on `main`.

File: src/client/amended.rs (hunk 1 above)
Comment (algesten) [reply src/client/amended.rs]: Fixed in c1b4761. The outgoing analyzers now use the single-value grammar (`parse_content_length_value`), so a comma separated list is rejected with `BadContentLengthHeader` before it can reach the wire. Both regression tests from 6dd1b37 pass, along with the full suite, fmt, clippy and docs.

File: src/server/amended.rs (hunk 2 above)
Comment (algesten) [reply src/server/amended.rs]: Fixed in c1b4761. The outgoing analyzers now use the single-value grammar (`parse_content_length_value`), so a comma separated list is rejected with `BadContentLengthHeader` before it can reach the wire. Both regression tests from 6dd1b37 pass, along with the full suite, fmt, clippy and docs.
